cmd/fetch.go

```diff
@@ -261,6 +275,30 @@ func fetchMesos(cmd *cobra.Command, args []string) {
 	fmt.Println(url)
 }
 
+func fetchMesosLeaderFromAgent(url string) (mesosLeaderHostName string) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return
+	}
+	if resp.StatusCode != 200 {
+		return
+	}
+	defer resp.Body.Close()
+
+	var res interface{}
+	err = json.NewDecoder(resp.Body).Decode(&res)
+	if err != nil {
+		return
+	}
+	hostname, ok := res.(map[string]interface{})["master_hostname"]
```

Comment:
Personal preference but I think it'd be better to create a struct type here.

Something like
```
type mesosAgent struct {
   MasterHostname string `json:master_hostname`
}
```
When you call decode on a type mesosAgent it'll automatically handle everything. That way you're not doing all this type assertion.

Reply:
Ah I thought I will need to create struct with complete JSON fields (and need to omit them explicitly). I will make this change.